frontend/packages/ceph-storage-plugin/src/components/ocs-storage-class-form.tsx

```diff
@@ -0,0 +1,174 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { K8sResourceKind } from '@console/internal/module/k8s';
+import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
+import { cephBlockPoolResource } from '../constants/resources';
+import { Dropdown } from '@patternfly/react-core/dist/js/components/Dropdown/Dropdown';
+import { DropdownItem } from '@patternfly/react-core/dist/js/components/Dropdown/DropdownItem';
+import { DropdownToggle } from '@patternfly/react-core/dist/js/components/Dropdown';
+import { CaretDownIcon } from '@patternfly/react-icons';
+
+const PoolResourceComponent: React.FC<PoolParams> = ({ onParamChange }) => {
+  const [data, loaded, loadError] = useK8sWatchResource<K8sResourceKind[]>(cephBlockPoolResource);
+  const [isOpen, setOpen] = React.useState<boolean>(false);
+  const [poolName, setPoolName] = React.useState('');
+
+  const handleDropdownChange = (e) => {
+    setPoolName(e.currentTarget.id);
+    onParamChange(e.currentTarget.id);
+  };
+
+  const poolDropdownItems = _.map(data, (pool, key) => (
+    <DropdownItem
+      key={key}
+      component="button"
+      id={pool?.metadata?.name}
```

Comment:
use getName function from shared console.

Reply:
> use getName function from shared console.

To be honest, I'm not a fan of that getter. It's not needed with optional chaining. I prefer it as in the PR.

Reply:
Ack! Any reason why? can we can convert the getter to use optional chaining & use it?